apiserver/facades/client/highavailability/highavailability_test.go

```diff
@@ -167,6 +187,46 @@ func (s *clientSuite) TestEnableHASeries(c *gc.C) {
 	c.Assert(machines[4].Series(), gc.Equals, "non-default")
 }
 
+func (s *clientSuite) TestEnableHAErrorForNoAddresses(c *gc.C) {
+	machines, err := s.State.AllMachines()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(machines, gc.HasLen, 1)
+	c.Assert(machines[0].Series(), gc.Equals, "quantal")
+
+	enableHAResult, err := s.enableHA(c, 3, emptyCons, defaultSeries, nil)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(enableHAResult.Added, gc.DeepEquals, []string{"machine-1", "machine-2"})
+
+	_, err = s.enableHA(c, 5, emptyCons, defaultSeries, nil)
+	c.Assert(err, gc.ErrorMatches,
+		`juju-ha-space is not set and a unique cloud-local address was not found for machines: \[1 2\]`)
```

Comment:
you can use err.Error(), gc.Equals if you find the regex stuff annoying.
It does give nicer error messages when things don't match.